### DISCUSSION TITLE
Simplify the return of a method avoiding response type declaration

### DISCUSSION THREAD

**Discussion by anibal-acosta** [dotnet/csharplang#5132] [Language Ideas] [answered]
Hello, I write a lot of code where a method return an object of type X, for example:

```
public PersonData GetPerson(int personId)
{
PersonData Response = new PersonData();
Response.Name = "John";
Response.Age = 30;
return Response;
}
```

I know that the above method can be simplyfied but is for example porpouse. What I suggest is... if return is used like object, compiler creates an object instance of the same type of the declared in the method construction (PersonData in this case). If return is not used like object, the behavior will be the same. 

```
public PersonData GetPerson(int personId)
{
return.Name = "John";
return.Age = 30;
}
```

**Comment by HaloFour** [comment, marked as answer]
I really don't like this style of function behavior.  It was how functions worked in VB5/6 and it was a near constant source of bugs due to accidental fallthrough.  In my opinion it's much better to be explicit about flow control.  Also, assuming `PersonData` is a reference type that also involves an implicit heap allocation (otherwise that pseudo `return` identifier would be `null`).

**Comment by koenbeuk** [comment]
Today we can already do:

```csharp
public PersonData GetPerson(int personId)  => new() { Name = "John", Age = 30  };
```

**Reply by FaustVX** [reply]
It's `=> new () { … };`
You have to write the `()`, otherwise, it's a creation of an anonymous type.

**Reply by koenbeuk** [reply]
Corrected, thanks!

**Reply by anibal-acosta** [reply]
for more complex cases will be some messy this but for simple escenarios it's ok.

**Reply by alfosua** [reply]
Well, it's not far from the example above:

```csharp
public PersonData GetPerson(int personId)  => new()
{
    Name = "John",
    Age = 30,
};
```

Although, I find more debuggable this:

```csharp
public PersonData GetPerson(int personId)
{
    PersonData result = new()
    {
        Name = "John",
        Age = 30,
    };
    return result;
}
```

Because you can watch the resulting object before return.

**Reply by jnm2** [reply]
> Because you can watch the resulting object before return.

I usually do this without creating a result variable by placing a breakpoint on the closing brace of the method and looking for the `GetPerson (return value)` line in the Locals window IIRC

**Reply by DavidArno** [reply]
@jnm2, I didn't know that was possible. Nice tip there. Thanks. 👍
